### PR TITLE
SEO ranking Plugin tests update

### LINKED ISSUE #17919: SEO Ranking plugin not working - problem with ping? - Fix the random test failures
Seeing this in the tests

image

and an error in the UI

image

Seems this regressed.

### PR DESCRIPTION
**Pull request by peterhashair**
### Description:

Fixes: #17919 

SEO test is failed randomly, normally a rerun will get it passed. 

Call a recluse to rerun the SEO test 3 times inside of the tests to reduce the chance of failing. 

Instead of test Microsoft, test matomo.org.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by peterhashair** [comment]
Not sure this will fix it, probably needs more people to run it in the Travis, I simulate run that test function 20 times in a short period, none of those simulations failed.

**Comment by tsteur** [comment]
@peterhashair is this one ready for review? We could otherwise simply merge and see if it still fails randomly over time. If we still see it failing then we'd try another approach.

**Comment by peterhashair** [comment]
@tsteur yeah, I think it still will fail over time, not a perfect solution, but should reduce the fail times.

File: plugins/SEO/tests/Integration/SEOTest.php [hunk 1]
```diff
@@ -60,4 +58,25 @@ public function provideContainerConfig()
...
+    {
+        if ($counter > 3) {
+            return [];
+        }
+        $dataTable = API::getInstance()->getRank('http://matomo.org/');
+        $renderer = Renderer::factory('json');
+        $renderer->setTable($dataTable);
+        $ranks = json_decode($renderer->render(), true);
+        foreach ($ranks as $rank) {
+            if ($rank['rank'] == 0 && $counter <= 3) {
```
Comment (justinvelluppillai) [inline plugins/SEO/tests/Integration/SEOTest.php]: `$counter <= 3` not needed here if you have a check above returning early. Also probably `===` be better for rank check

File: plugins/SEO/tests/Integration/SEOTest.php [hunk 2]
```diff
@@ -41,10 +41,8 @@ public function setUp(): void
      */
     public function test_API()
     {
-        $dataTable = API::getInstance()->getRank('http://www.microsoft.com/');
-        $renderer = Renderer::factory('json');
-        $renderer->setTable($dataTable);
-        $ranks = json_decode($renderer->render(), true);
+        $ranks = $this->apiFunction();
+        $this->assertNotEmpty($ranks, 'Exceed 3 times, maximum tried');
```
Comment (justinvelluppillai) [inline plugins/SEO/tests/Integration/SEOTest.php]: This error message will be a bit less helpful - we won't know which one failed whereas before it would tell us bing.

**Review by justinvelluppillai — APPROVED** [review]
Looks good to merge once tests have run.

**Comment by peterhashair** [comment]
@tsteur maybe that doesn't work. see here https://app.travis-ci.com/github/matomo-org/matomo/jobs/542068646

**Comment by tsteur** [comment]
👍  I'll reopen the issue again so we can try a different approach
